signer/xpi/x509.go

```diff
@@ -117,12 +118,69 @@ func (s *XPISigner) makeTemplate(cn string) *x509.Certificate {
 	}
 }
 
+// getIssuerPubKey returns the public key for ECDSA or RSA keys from
+// the crypto stdlib or crypto11
+func (s *XPISigner) getIssuerPubKey() (pubKey crypto.PublicKey, err error) {
```

Comment:
This and the following three functions could move to signer.

Reply:
Yeah, why not use the existing `signer.GetKeysAndRand` instead, and store the pubkey in the `XPISigner`?